File: talleres c++/funciones.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    // Definición de los vectores
    vector<int> a = {1, 2, 3, 4, 5};
    vector<int> b = {6, 7, 8, 9, 10};
    vector<int> c;

    // Multiplicación de los elementos de 'a' y 'b'
    for (size_t contador = 0; contador < a.size(); ++contador) {
        c.push_back(a[contador] * b[contador]);
    }

    // Imprimir el resultado de la multiplicación
    cout << "Resultado de multiplicación de a y b: ";
    for (int val : c) {
        cout << val << " ";
    }
    cout << endl;

    // Funciones

    // Definición y llamada de función
    auto mostrar_texto = []() {
        cout << "Hola" << endl;
    };

    mostrar_texto();

    // Función que multiplica dos números
    auto multiplicar = []() {
        int a = 7;
        int b = 9;
        cout << a * b << endl;
    };

    multiplicar();

    // Ámbito de las variables
    int a = 7, b = 9;
    auto multiplicar2 = [a, b]() { // Captura las variables a y b por valor
        cout << a * b << endl;
    };

    multiplicar2();

    // Función que devuelve un valor y luego se manipula en la variable fuera de la función
    auto multiplicar3 = []() -> int {
        int a = 7;
        int b = 9;
        return a * b;
    };

    int Resultado = multiplicar3();
    cout << Resultado + 5 << endl;

    // Función que valida un dato
    auto validar_dato = []() -> bool {
        int a = 9;
        if (a == 7) {
            return true;
        } else {
            return false;
        }
    };

    bool dato = validar_dato();
    cout << (dato ? "true" : "false") << endl;

    return 0;
}

File: RUBI/funciones.rb
# Multiplicación de elementos de dos tuplas (arrays en Ruby)
a = [1, 2, 3, 4, 5]
b = [6, 7, 8, 9, 10]
c = []

# Multiplicación de elementos correspondientes de las listas
a.each_with_index do |valor, contador|
  c.append(valor * b[contador])
end
puts c

# Funciones en Ruby

# Definición y llamada de función
def mostrar_texto
  puts 'hola'
end

mostrar_texto

# Función multiplicar
def multiplicar
  a = 7
  b = 9
  puts a * b
end

multiplicar

# Ámbito de las variables
def multiplicar
  puts a * b
end

a = 7
b = 9
multiplicar

# Función que devuelve un valor
def multiplicar
  a = 7
  b = 9
  return a * b
end

resultado = multiplicar
puts resultado + 5

# Función que valida un dato
def validar_dato
  if a == 7
    return true
  else
    return false
  end
end

a = 9
dato = validar_dato
puts dato
